src/fudge/log.clj

```diff
@@ -118,40 +130,62 @@
 
 ;; # Main logging functions
 
-(defn log
-  "Log a record with the config, level, and data provided"
-  [c level data]
-  (let [record (invoke c :prepare-fn data level)]
+(defn log!
+  "Log a record with the config, level, form metadata and data provided"
+  [c calling-meta level data]
+  (let [record (invoke c :prepare-fn data level calling-meta)]
     (when (invoke c :log?-fn record)
       (->> record
            (call c :format-fn)
            (call c :output-fn)))))
 
-(defn spy-with
+(defmacro log
+  "Log a record with the config, level, and data provided."
+  [c level data]
+  `(log! ~c (calling-meta) ~level ~data))
+
+(defmacro spy-with->>
```

Comment:
Good spot re `->>`.  I wonder if we want an alias without the punctuation noise though.

Also, why a macro?